### PR TITLE
Update AppArmor validator to use up to date host information

### LINKED ISSUE #116003: AppArmor validator: Use up to date host info
This issue is created by this @SergeyKanzhelev's [request](https://github.com/kubernetes/kubernetes/pull/115749#issuecomment-1441305000)

Currently AppArmor validator gets host info only [when Validator object is created](https://github.com/kubernetes/kubernetes/blob/master/pkg/security/apparmor/validate.go#L44).
`Validator.ValidateHost` API [uses cached data](https://github.com/kubernetes/kubernetes/blob/master/pkg/security/apparmor/validate.go#L85).

Would it make sense to do it dynamically every time ValidateHost is called? It would help to get the latest AppArmor status, which can be changed if AppArmor kernel module is loaded or unloaded.

### PR DESCRIPTION
**Pull request by j0shgrant**
/kind feature
/kind api-change
/sig node
/sig security

#### What this PR does / why we need it:

This PR updates the AppArmor validator to validate host configuration on every pod validation (instead of caching any host configuration error encountered at kubelet startup, then returning that error for any subsequent `Validate` calls). This allows the validator to use the host's latest AppArmor configuration, which may have changed if the kernel module was loaded/unloaded.

#### Which issue(s) this PR fixes:

Fixes #116003

#### Does this PR introduce a user-facing change?
```release-note
AppArmor configuration changes caused by (un)loading the kernel module will be reflected in any subsequent validations performed by the AppArmor pod admission controller.
```

### CODE REVIEW

**Review by rasooll — APPROVED** [review]

File: pkg/security/apparmor/shim.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// Shim interface allows injection of bad host configuration for testing purposes
+type Shim interface {
+	IsEnabled() bool
+}
+
+type shim struct{}
+
+func (s shim) IsEnabled() bool {
+	return apparmor.IsEnabled()
+}
```
Comment (bart0sh) [inline pkg/security/apparmor/shim.go:30]: Do we really need to put this code into a separate file?

Comment (j0shgrant) [reply pkg/security/apparmor/shim.go]: Not really - moved it into `validate.go` for simplicity.

**Comment by tallclair** [comment]
/hold
For decision on whether we want this change, being discussed on https://github.com/kubernetes/kubernetes/issues/116003

**Comment by SergeyKanzhelev** [comment]
Replaced by https://github.com/kubernetes/kubernetes/pull/116606

/close

thank you, @j0shgrant!

**Pull request closed without merging**